dell/src/main/java/org/apache/iceberg/dell/ecs/EcsCatalog.java

```diff
@@ -0,0 +1,520 @@
+    return Namespace.of(namespace);
+  }
+
+  /**
+   * Load namespace properties.
+   */
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) throws NoSuchNamespaceException {
+    EcsURI namespaceObject = namespaceURI(namespace);
+    if (!objectMetadata(namespaceObject).isPresent()) {
+      throw new NoSuchNamespaceException("Namespace %s(%s) properties object is absent", namespace, namespaceObject);
+    }
+
+    Map<String, String> result = loadProperties(namespaceObject).content();
+
+    LOG.debug("Loaded metadata for namespace {} found {}", namespace, result);
+    return result;
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyException {
+    if (!namespace.isEmpty() && !namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist", namespace);
+    }
+
+    if (!listNamespaces(namespace).isEmpty() || !listTables(namespace).isEmpty()) {
+      throw new NamespaceNotEmptyException("Namespace %s is not empty", namespace);
+    }
+
+    EcsURI namespaceObject = namespaceURI(namespace);
+    client.deleteObject(namespaceObject.bucket(), namespaceObject.name());
+    LOG.info("Dropped namespace: {}", namespace);
+    return true;
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) throws NoSuchNamespaceException {
+    return updateProperties(namespace, r -> r.putAll(properties));
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> properties) throws NoSuchNamespaceException {
+    return updateProperties(namespace, r -> r.keySet().removeAll(properties));
+  }
+
+  public boolean updateProperties(Namespace namespace, Consumer<Map<String, String>> propertiesFn)
+      throws NoSuchNamespaceException {
+
+    // Load old properties
+    Properties oldProperties = loadProperties(namespaceURI(namespace));
+
+    // Put new properties
+    Map<String, String> newProperties = new LinkedHashMap<>(oldProperties.content());
+    propertiesFn.accept(newProperties);
+    LOG.debug("Successfully set properties {} for {}", newProperties.keySet(), namespace);
+    return updatePropertiesObject(namespaceURI(namespace), oldProperties.eTag(), newProperties);
+  }
+
+  @Override
+  public boolean namespaceExists(Namespace namespace) {
+    return objectMetadata(namespaceURI(namespace)).isPresent();
+  }
+
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    return objectMetadata(tableURI(identifier)).isPresent();
+  }
+
+  private void checkURI(EcsURI uri) {
+    Preconditions.checkArgument(uri.bucket().equals(dellProperties.ecsCatalogPrefix().bucket()),
+        "Properties object %s should be in same bucket", uri.location());
+    Preconditions.checkArgument(uri.name().startsWith(dellProperties.ecsCatalogPrefix().name()),
+        "Properties object %s should have prefix", uri.location());
```

Comment:
`Should have the expected prefix %s`, It's more friendly to add this ecs catalog prefix.

Reply:
Fixed.